### PR TITLE
Support for an `allowedEndpoints` option for `plugin-proxy`.

### PR DESCRIPTION
**Pull request by dobs**
## Hey, I just made a Pull Request!

This enables restricting proxy requests to a specified set of endpoints and associated HTTP verbs.

As an example use case, currently plugins that rely on the proxy plugin (e.g. `plugin-pagerduty`) open up unfettered access to the server being proxied. Strict enforcement might involve a corresponding backend plugin rather than using the proxy at all, allow-listing can at least restrict users to a strict set of calls.

The current implementation is strict: The path must match exactly. No globs, no regex. Query params may vary, though.

Feedback welcome -- there are a few different ways the allow list could be represented and I chose one somewhat arbitrarily. :)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by dobs** [comment]
On documentation -- the proxy docs appear to currently live outside the project?

Happy to submit changes if there's another project I should be looking at for the docs, or can include some content in comments here.

File: .changeset/spotty-insects-heal.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-proxy-backend': minor
+---
+
+Support for an `allowedEndpoints` option for `plugin-proxy`.
```
Comment (Fox32) [inline .changeset/spotty-insects-heal.md:5]: A think that comes into my mind: you need to add the new property to the config schema

**Comment by dhenneke** [comment]
> On documentation -- the proxy docs appear to currently live outside the project?
> 
> 
> 
> Happy to submit changes if there's another project I should be looking at for the docs, or can include some content in comments here.

If you mean the proxy docs that is part of [backstage.io/docs](https://backstage.io/docs/plugins/proxying), you can edit them here: https://github.com/backstage/backstage/blob/master/docs/plugins/proxying.md

File: .changeset/spotty-insects-heal.md (hunk 1 above)
Comment (dobs) [reply .changeset/spotty-insects-heal.md:5]: Thanks! Adding that now. :)

**Comment by dobs** [comment]
@dhenneke Thanks! I wasn't looking hard enough -- will update those as well. :)

File: .changeset/spotty-insects-heal.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-proxy-backend': minor
```
Comment (benjdlambert) [inline .changeset/spotty-insects-heal.md:2]: Think this is still a `patch` change? Whilst we are `<1.0.0` `minor === patch`

**Comment by benjdlambert** [comment]
Wondering if the extent of these changes are outside of the scope for the proxy plugin, and that anything that requires this sort of feature should be converted into a backend plugin instead.

We appreciate the contribution, but we also realise that this plugin is a awash with security holes that could be exploited with the tokens too. 

Thinking that instead of enabling this functionality, we'd just advise against it, and encourage writing backend plugins instead. 

The end goal is that we can also replace the proxy with some form of GraphQL Queries and Mutations, so that it can take care of abstracting away the calls to API's and limit what you can and can't do.

**Comment by dobs** [comment]
@benjdlambert That's good feedback, thanks!

Having a custom backend was something I'd been considering as a longer-term plan, so I'd be OK skipping the proxy changes if you don't think they'd be more broadly useful.

Closing for the time being but happy to re-open and make the previously-requested changes if/when there's interest. :)

**Pull request closed without merging**